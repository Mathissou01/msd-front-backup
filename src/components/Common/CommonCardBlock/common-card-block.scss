@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/02-tools/tools.shadows" as shadows;

.c-CommonCardBlock {
  $self: &;

  display: flex;
  flex: 1;
  flex-direction: column;
  gap: 16px;
  padding-bottom: 24px;

  @include im.media(">=mobile") {
    gap: 24px;
  }

  &:hover {
    background-color: colors.$brand-primary-light;
  }

  &_isEventDisplay {
    padding: 24px 16px;
  }

  &__Container {
    @include shadows.shadow(1);

    display: flex;
    flex-direction: column;
    width: 100%;
    overflow: hidden;
    background-color: colors.$white;
    border-radius: 8px;
  }

  &__EventDisplay {
    position: relative;
    display: flex;
    flex-direction: column;
    gap: 8px;
    align-items: center;
    justify-content: center;
    width: 132px;
    height: 132px;
    margin: auto;
    padding: 32px;
    background: rgb(244 154 28 / 10%);
    border-radius: 50%;
  }

  &__CalendarPicto {
    width: 36px;
    filter: url("#recolor-secondary");
  }

  &__Date {
    width: 150px;
    color: colors.$gray-800;
    font-weight: 500;
    font-size: 14px;
    font-family: DINPro-Medium, sans-serif;
    line-height: 21px;
    text-align: right;
  }

  &__Image {
    width: 100%;
    height: 100%;
    overflow: hidden;

    @include im.media(">=tablet") {
      height: 200px;
      min-height: 200px;
    }

    img {
      width: 100%;
      height: 100%;
      transition: transform 0.2s ease-in-out;

      @include im.media(">=tablet") {
        object-fit: cover;
      }

      #{$self}:hover & {
        transform: scale(1.1);
      }
    }
  }

  &__Content {
    display: flex;
    flex-direction: column;
    gap: 16px;
    height: 100%;
    padding: 0 16px;

    &_textCenter {
      align-items: center;
      text-align: center;
    }
  }

  &__ContentHeader {
    display: flex;
    gap: 16px;
    align-items: center;
    justify-content: space-between;
  }

  &__Tags {
    display: flex;
    flex-wrap: wrap;
    gap: 6px;
    align-items: center;
    max-width: 175px;
  }

  &__Tag {
    @include typo.text("tag");

    padding: 6px 10px;
    border: 1px solid colors.$blue-4;
    border-radius: 4px;

    &_background {
      background-color: colors.$gray-0;
      border: hidden;
    }
  }

  &__ContentBody {
    display: flex;
    flex-direction: column;
    gap: 6px;
    padding: 0;
  }

  &__Title {
    @include typo.heading(3);

    width: 100%;
    color: colors.$expert-blue;
  }

  &__Description {
    color: colors.$expert-blue;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
  }

  &__Link {
    margin-top: auto;
    padding: 0;
    transition: text-decoration 0.3s ease-in-out;

    #{$self}:hover & {
      text-decoration: underline;
    }

    span {
      @include typo.text("buttonSmall");

      margin-right: 12px;
      color: colors.$blue-1;
      border-radius: 2px;
    }
  }
}
