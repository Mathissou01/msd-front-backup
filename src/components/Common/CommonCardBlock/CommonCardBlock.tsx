import classNames from "classnames";
import Image from "next/image";
import Link from "next/link";
import React from "react";
import {
  TagEntity,
  UploadFile,
} from "../../../graphql/codegen/generated-types";
import {
  handleDateFrenchFormat,
  makePublicAssetPath,
} from "../../../lib/utilities";
import "./common-card-block.scss";

interface ICommonCardBlockProps {
  href: string;
  title: string;
  shortDescription?: string;
  date?: string;
  tags?: Array<TagEntity>;
  image?: UploadFile | null;
  isAlignTextCenter?: boolean;
  isEventDisplay?: boolean;
}

export default function CommonCardBlock({
  href,
  title,
  shortDescription,
  date,
  tags,
  image,
  isEventDisplay = false,
  isAlignTextCenter = false,
}: ICommonCardBlockProps) {
  /* Static Data */
  const linkLabel = "En savoir plus";
  const contentCardDate = new Date(date ?? "");
  const dataFrenchFormat = handleDateFrenchFormat(contentCardDate);
  const arrowIcon = {
    source: "images/pictos/arrow.svg",
    alternativeText: "",
    ariaHidden: true,
  };
  const blockClasses = classNames("c-CommonCardBlock", {
    "c-CommonCardBlock_isEventDisplay": isEventDisplay,
  });
  const contentClasses = classNames("c-CommonCardBlock__Content", {
    "c-CommonCardBlock__Content_textCenter": isAlignTextCenter,
  });

  return (
    <Link className="c-CommonCardBlock__Container" href={href}>
      <div className={blockClasses}>
        {isEventDisplay ? (
          <div className="c-CommonCardBlock__EventDisplay">
            <Image
              className="c-CommonCardBlock__CalendarPicto"
              src={makePublicAssetPath("/images/pictos/calendar.svg")}
              alt={""}
              width={36}
              height={36}
            />
            {date && (
              <div className="c-CommonCardBlock__Date">
                {dataFrenchFormat ?? null}
              </div>
            )}
          </div>
        ) : (
          <div className="c-CommonCardBlock__Image">
            {image?.url && (
              <Image
                src={makePublicAssetPath(image.url)}
                alt={image?.alternativeText ?? ""}
                width={image?.width ?? 482}
                height={image?.height ?? 309}
              />
            )}
          </div>
        )}
        <div className={contentClasses}>
          <div className="c-CommonCardBlock__ContentHeader">
            <div className="c-CommonCardBlock__DateContainer">
              {dataFrenchFormat}
            </div>
            {tags?.map((tagLabel, index) => (
              <span
                key={index}
                className="c-CommonCardBlock__Tag c-CommonCardBlock__Tag_background"
              >
                {tagLabel.attributes?.name}
              </span>
            ))}
          </div>
          <div className="c-CommonCardBlock__ContentBody">
            <h3 className="c-CommonCardBlock__Title">{title}</h3>
            <p className="c-CommonCardBlock__Description">{shortDescription}</p>
          </div>
          <div className="c-CommonCardBlock__Link">
            <span>{linkLabel}</span>
            <Image
              src={makePublicAssetPath(arrowIcon.source)}
              alt={arrowIcon.alternativeText}
              aria-hidden={arrowIcon.ariaHidden}
              width={24}
              height={24}
            />
          </div>
        </div>
      </div>
    </Link>
  );
}
