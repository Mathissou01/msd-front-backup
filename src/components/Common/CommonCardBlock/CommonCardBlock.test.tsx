import { render } from "@testing-library/react";
import CommonCardBlock from "./CommonCardBlock";

const mock = {
  tagLabels: ["common card block tag"],
  title: "common card block title",
  href: "/accueil",
  image: {
    url: "/images/images-temp/temp_image.jpg",
    alternativeText: "",
    hash: "",
    mime: "",
    name: "",
    provider: "",
    size: 0,
  },
};

describe("CommonCardBlock", () => {
  it("renders", () => {
    const { container } = render(
      <CommonCardBlock
        tagLabels={mock.tagLabels}
        title={mock.title}
        image={mock.image}
        href={mock.href}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
