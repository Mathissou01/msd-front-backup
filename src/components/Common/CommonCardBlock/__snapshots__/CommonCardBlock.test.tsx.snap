// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonCardBlock renders 1`] = `
<div>
  <a
    class="c-CommonCardBlock__Container"
    href="/accueil"
  >
    <div
      class="c-CommonCardBlock"
    >
      <div
        class="c-CommonCardBlock__Image"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="309"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=1080&q=75"
          srcset="/_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=1080&q=75 2x"
          style="color: transparent;"
          width="482"
        />
      </div>
      <div
        class="c-CommonCardBlock__Content"
      >
        <div
          class="c-CommonCardBlock__ContentHeader"
        >
          <div
            class="c-CommonCardBlock__DateContainer"
          >
            Invalid Date
          </div>
          <span
            class="c-CommonCardBlock__Tag c-CommonCardBlock__Tag_background"
          >
            Prise de RDV
          </span>
          <span
            class="c-CommonCardBlock__Tag c-CommonCardBlock__Tag_background"
          >
            Ev√©nement
          </span>
        </div>
        <div
          class="c-CommonCardBlock__ContentBody"
        >
          <h3
            class="c-CommonCardBlock__Title"
          >
            common card block title
          </h3>
          <p
            class="c-CommonCardBlock__Description"
          />
        </div>
        <div
          class="c-CommonCardBlock__Link"
        >
          <span>
            En savoir plus
          </span>
          <img
            alt=""
            aria-hidden="true"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/images/pictos/arrow.svg"
            style="color: transparent;"
            width="24"
          />
        </div>
      </div>
    </div>
  </a>
</div>
`;
